compiler/rustc_span/src/symbol.rs

```diff
@@ -749,11 +759,17 @@ symbols! {
         extern_types,
         external_doc,
         f,
-        f16c_target_feature,
+        f128,
+        f128_nan,
+        f16,
+        f16_nan,
+        f16c_target_feature, // todo: how should this interact?
```

Comment:
I don't know what exactly this feature is for, or how it needs to interact with `f16`

Reply:
@tgross35 This is about support for `target_feature` for the x86(_64) F16C extensions, which introduces hardware support for f16 conversions.  In other words, it's conceptually related but practically independent of this, since handling it is LLVM's job and rustc just needs to know it's there.

Reply:
This feature doesn't affect how f16/f128 is passed across function boundaries, or does it?

Reply:
@RalfJung It shouldn't?  We should be passing f16 and f128 via xmm registers always, right?  …right?  😟

Reply:
I have no idea. Someone should check that. After all xmm registers are not available without sse, e.g.  on i586 targets.

Reply:
This isn't an issue from an architectural standpoint since F16C presupposes SSE4.2 (and then some) but it might still be an ABI issue if a calling convention doesn't want to use those registers for whatever reason.

Reply:
So what happens when one writes this code without SSE enabled? (or with SSE explicitly disabled)
```rust
pub fn test(x: f16) -> f16 { x }
```
This doesn't even need any f16 operations. Though of course the same question arises with
```rust
pub fn test(x: f16) -> f16 { x + 1.0 }
```

Reply:
> So what happens when one writes this code without SSE enabled? (or with SSE explicitly disabled)
> 
> ```rust
> pub fn test(x: f16) -> f16 { x }
> ```

My first impression would be that it gets passed in the GPRs, like other soft-float ABIs, even if f16c is an available extension.  So maybe it's not a problem after all?

Reply:
This definitely needs to be checked before stabilization. We'll want to make sure that the values get passed the same way *no matter which target features are enabled*.

See https://github.com/rust-lang/lang-team/issues/235 for context.

Reply:
Well, wound up stumbling across this by accident https://github.com/rust-lang/rust/issues/123885